include/posix/sys/time.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_POSIX_SYS_TIME_H_
+#define ZEPHYR_INCLUDE_POSIX_SYS_TIME_H_
+
+#ifdef CONFIG_NEWLIB_LIBC
+/* Kludge to support outdated newlib version as used in SDK 0.10 for Xtensa */
+#include <newlib.h>
+
+#ifdef __NEWLIB__
+#include <sys/_timeval.h>
```

Comment:
Are you expecting more things to end up in _timeval.h?  Just curious why to introduce a new file instead of just putting the struct here directly.

Reply:
We discussed that at Connect and commit message explains - I'm introduce that file because newlib has that file with that define. It gets included by various other newlib headers. If we have concurrent definition in some other file (like Zephyr POSIX header), we get conflict. We should not play against newlib, but along with it, reusing its definitions. But then the other libc (minimal) should follow the same file structuring and naming, or we'll never get all the different configurations to work consistently.

Now why newlib put that declaration in a separate is a different matter. I can theorize that as mentioned, `struct timeval` declaration is needed in various headers, e.g. select.h. But nowhere POSIX would say something like "including select.h should have the effect of including sys/time.h". That's why `struct timeval` is factored out to a separate (libc-private) header, and fine-grained inclusion happens.